### PR TITLE
Removes /datum/topic_input, a helper datum for assoc lists

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
I don't understand why it exists. Anything that uses it also makes raw accesses into the list anyways. Nothing it does can't be done better with a define, but also pretty sure all the use cases are broken and buggy code with issues if you just look.

Mecha code needs a rewrite.

### CODE REVIEW

File: code/game/mecha/mecha_topic.dm [hunk 1]
```diff
@@ -216,43 +216,48 @@
...
-		if(href_list["req_access"] && add_req_access)
-			output_access_dialog(afilter.getObj("id_card"),afilter.getMob("user"))
+		if(href_list["req_access"] && add_req_access && user && id_card)
+			output_access_dialog(id_card,user)
 
-		if(href_list["maint_access"] && maint_access)
-			var/mob/user = afilter.getMob("user")
-			if(user)
+		if(href_list["maint_access"] && maint_access && user && id_card)
+			if (istype(id_card))
```
Comment (SpaceManiac) [inline code/game/mecha/mecha_topic.dm]: this `istype` is redundant with the earlier one, I think

File: code/game/mecha/mecha_topic.dm [hunk 2]
```diff
@@ -216,43 +216,48 @@
 	if(usr.incapacitated())
 		return
 
-	var/datum/topic_input/afilter = new /datum/topic_input(href,href_list)
-
 	if(in_range(src, usr))
+		var/mob/user
+		var/obj/item/card/id/id_card
+		if (href_list["user"])
```
Comment (SpaceManiac) [inline code/game/mecha/mecha_topic.dm]: Maybe out of scope but why in the world is this part of the topic call, there is an exploit here to basically show popup dialogs to anyone you can get the REF of. Should just be `usr`.

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
yikes @ mecha code

**Comment by vuonojenmustaturska** [comment]
agreed on "yikes"

**Pull request merged**